### DISCUSSION TITLE
How to test SelectionList.SelectionToggled event?

### DISCUSSION THREAD

**Discussion by jvllmr** [Textualize/textual#4397] [Q&A] [answered]
In my app I use the `SelectionList.SelectionToggled` event for reacting to SelectionList option changes. 
The event only fires on user interaction, but not by API usage.
This means to me that I need to use `Pilot.click` which doesn't work because SelectList Options are not rendered as an element with an id and you therefore cannot fetch the object via a query.

Anyone knows a way around this? (without some weird offset magic or synthesizing the event)

**Comment by TomJGooding** [comment]
If the options aren't widgets, I don't see how else it would be possible with pilot.click without using offset ?

> The event only fires on user interaction, but not by API usage.

If you're referring to the note in the docs, [I _think_ this might be out of date,](https://github.com/Textualize/textual/issues/4398)

**Reply by jvllmr** [reply, marked as answer]
Alright, now I know where I went wrong: When looking into textual's code again I found out that `SelectionList.toggle` triggers the event while `SelectionList.select` does not. I only tried out `SelectionList.select` before :(
